chanfitness/chanevent.go

```diff
@@ -212,7 +267,7 @@ type onlinePeriod struct {
 // online or offline events.
 func (p *peerLog) getOnlinePeriods() []*onlinePeriod {
 	// Return early if there are no events, there are no online periods.
-	if len(p.onlineEvents) == 0 {
+	if len(p.listEvents()) == 0 {
```

Comment:
nit: could save the result of `listEvents` to a variable to avoid creating the list twice.